modules/configure-with-cli.md

```diff
@@ -0,0 +1,204 @@
+---
+title: Configure with CLI
+weight: 0
+---
+
+# Configure a module with CLI
+
+From {{< minver v="1.7.2" >}}, Self Configurator feature was introduced in the `ModuleCommand`.
+
+Self Configurator is a feature that allows module configuration management from a command, without having to connect on the back office of the shop.
+
+From a configuration file, the command is able to do some configuration on a specific module, e.g.:
+
+- change values in the `Configuration` table
+- execute SQL statements from a `.sql` file
+- copy files
+- execute `php` scripts
+
+Complete reference of Self Configurator: [ModuleSelfConfigurator.php](https://github.com/PrestaShop/PrestaShop/blob/8.0.x/src/Adapter/Module/Configuration/ModuleSelfConfigurator.php)
+
+## How to use it
+
+You need to create a configuration file, with the `yaml` [format](https://symfony.com/doc/current/components/yaml/yaml_format.html), and execute:
+
+`php bin/console prestashop:module configure <modulename> <configfilepath>`
+
+{{% notice info %}}
+Please note that the third argument `<configfilepath>` is optional. If not set, PrestaShop will try to find a file named `self_config.yml` in the module's folder (and sub-folders).
+{{% /notice %}}
+
+## Apply configuration file on a specific shop or group
+
+The Prestashop `prestashop:module` command allows you to restrict its actions on a specific shop (or group).
+
+To do so, add an `--id_shop` or an `--id_shop_group` argument to the command:
+
+```bash
+php bin/console prestashop:module configure <modulename> <configfilepath> --id_shop=<id>
+php bin/console prestashop:module configure <modulename> <configfilepath> --id_shop_group=<id>
+```
+
+## Configuration file reference
+
+### Change values in the Configuration table
+
+The Self Configurator feature allows editing values in the `Configuration` table from the configuration file. 
+
+To do so, add a `configuration:` line, choose if you want to update or delete a configuration value, and add the configuration key and its value like in those examples:
+
+```yaml
+configuration:
+    update:
+        PAYPAL_SANDBOX: 1
+```
+
+```yaml
+configuration:
+    delete:
+        - "PAYPAL_ONBOARDING"
+```
+
+{{% notice info %}}
+Please note that you can only `update` or `delete` values: `create` values are only done when installing the module.
```

Comment:
You can add value. The update will create if needed.

Reply:
thanks, fixed 👍